api/handlers/changes.js

```diff
@@ -208,25 +205,33 @@ var getChanges = function(feed) {
     },
     (err, responses) => {
       endTimer(`getChanges().requests:${chunks.length}`, startTime);
-
-      if (feed.res.finished) {
-        // Don't write to the response if it has already ended. The change
-        // will be picked up in the subsequent changes request.
-        return;
-      }
-      cleanUp(feed);
       if (err) {
-        feed.res.write(error(503, 'Error processing your changes'));
-      } else {
+        return errorResponse(feed);
+      }
+
+      // if relevant ids have changed, update validated ids, getChanges again
+      const originalValidatedIds = feed.validatedIds;
+      bindServerIds(feed, err => {
```

Comment:
I thought you were renaming all the 'bind' functions?

Reply:
So this is adding an additional `bindServerIds` call into every changes feed handle?

Reply:
I changed _most_ of the bind functions so that instead of binding directly on to the `feed` object they return the data to bind and so I changed their name from "bindXYZ" to "getXYZ". This is (I think) the last one and it's the one that calls all the other getters and stores the result on the `feed` so it's the one that truly binds.

Yes we now do four database view requests when the changes handler fires. This is clearly a performance regression but it's the only way I could figure out to make sure no changes are missed that doesn't require a rewrite. I believe this is an acceptable compromise for now as I don't want to introduce instability in a 2.14.x patch. We should still undertake the rewrite to fix #4172 as well as other improvements that are overdue.